build.sbt

```diff
@@ -18,6 +18,7 @@ lazy val scalaLibraryNext = crossProject(JVMPlatform, JSPlatform)
   .jsEnablePlugins(ScalaJSJUnitPlugin)
   .settings(
     ScalaModulePlugin.scalaModuleSettings,
+    versionPolicyIntention := Compatibility.None, // TODO Change to `Compatibility.BinaryAndSourceCompatible` after the first release
```

Comment:
we're committing to source compatibility? that seems... unusual

Reply:
We relax the compatibility guarantees according to our needs. In the case of scala-library-next, we never want break the binary compatibility, so we will be using `Compatibility.BinaryAndSourceCompatible` between patch releases, and `Compatibility.BinaryCompatible` between minor releases.

Reply:
see https://github.com/scalacenter/sbt-version-policy/issues/71 for discussion on what "source compatibility" even means in this context

Reply:
are we using `BinaryAndSourceCompatible` as a proxy for forward and backward binary compatibility here? this seems like we've gone in a very confusing circle

Reply:
> are we using `BinaryAndSourceCompatible` as a proxy for forward and backward binary compatibility here?

No. I am sorry if this is confusing, but we are using `BinaryAndSourceCompatible` between patch releases (which are expected to be both binary and source compatible with previous releases), and we use `BinaryCompatible` only between minor releases (which are expected to be binary compatible with previous releases). Does that make sense? I don’t understand what is problematic here.

Reply:
I didn't think we committed to strict source compatibility between patch releases is all

Reply:
@NthPortal Yeah, according to the [recommended versioning scheme](https://docs.scala-lang.org/overviews/core/binary-compatibility-for-library-authors.html#recommended-versioning-scheme), patch releases must not break compilation (in addition to being binary compatible with the previous patch releases), so they must be source compatible.